classes/api/package_api.php

```diff
@@ -156,6 +163,94 @@ public function score_attempt(string $questionstate, string $attemptstate, ?stri
         return array_converter::from_array(attempt_scored::class, $httpresponse->get_data());
     }
 
+    /**
+     * Send a POST request and retry if the server doesn't have the package file cached, but we have it available.
+     *
+     * @param string $uri can be absolute or relative to the base url
+     * @param array $options request options as per
+     *                       {@link https://docs.guzzlephp.org/en/stable/request-options.html Guzzle docs}
+     * @param bool $allowretry if set to false, retry won't be attempted if the package file isn't cached, instead
+     *                         throwing a {@see coding_exception}
+     * @return ResponseInterface
+     * @throws coding_exception if the request is unsuccessful for any other reason
+     * @see post_and_maybe_retry
+     */
+    private function guzzle_post_and_maybe_retry(string $uri, array $options = [], bool $allowretry = true): ResponseInterface {
+        try {
+            return $this->client->post($uri, $options);
+        } catch (BadResponseException $e) {
+            if (!$allowretry || !$this->file || $e->getResponse()->getStatusCode() != 404) {
+                throw $e;
+            }
+
+            try {
+                $json = Utils::jsonDecode($e->getResponse()->getBody(), assoc: true);
+            } catch (InvalidArgumentException) {
+                // Not valid JSON, so the problem probably isn't a missing package file.
+                throw $e;
+            }
```

Comment:
Da wir die Fehlermeldung nicht brauchen, ginge hier auch direkt:
```suggestion
            $json = json_decode($e->getResponse()->getBody(), associative: true);
            if (JSON_ERROR_NONE !== json_last_error()) {
                // Not valid JSON, so the problem probably isn't a missing package file.
                throw $e;
            }
```